R/status_addin.R

```diff
@@ -0,0 +1,98 @@
+#' Addin for viewing status of orderly develop mode.
+#'
+#' Allows users to reload dependencies and to cleanup development mode.
+#'
+#' @export
+status_addin <- function() {
+
+  # Our ui will be a simple gadget page, which
+  # simply displays the time in a 'UI' output.
+  ui <- miniUI::miniPage(
+    miniUI::miniContentPanel(
+        DT::dataTableOutput("status")
+    ),
+    miniUI::miniButtonBlock(
+      shiny::actionButton("refresh", "Reload dependencies",
+                          shiny::icon("redo")),
+      shiny::actionButton("clean", "Cleanup develop mode",
+                          shiny::icon("broom")),
+      shiny::actionButton("done", "Close",
+                          shiny::icon("times"))
+    )
+  )
+
+  server <- function(input, output, session) {
+
+    status <- orderly::orderly_develop_status()
+    output$status <- render_status(status)
+
+    shiny::observeEvent(input$row, {
+      file <- status[input$row, "filename"]
+      if (file.exists(status[input$row, "filename"])) {
+        rstudioapi::navigateToFile(status[input$row, "filename"])
+      } else {
+        shiny::showNotification(sprintf("File %s does not exist", file),
+                                type = "message")
+      }
+    })
+
+    # Listen for 'done' events. When we're finished, we'll
+    # stop the gadget.
+    shiny::observeEvent(input$done, {
+      shiny::stopApp()
+    })
+
+    ## List for refresh button click
+    shiny::observeEvent(input$refresh, {
+      tryCatch({
+        orderly::orderly_develop_start()
+        status <- orderly::orderly_develop_status()
+        output$status <- render_status(status)
+      },
+      error = function(e) {
+        message(e$message)
+        shiny::showNotification(e$message, type = "error")
+      })
+    })
+
+    ## List for clean button click
+    shiny::observeEvent(input$clean, {
+      tryCatch({
+        orderly::orderly_develop_clean()
+        status <- orderly::orderly_develop_status()
+        output$status <- render_status(status)
+      },
+      error = function(e) {
+        message(e$message)
+        shiny::showNotification(e$message, type = "error")
+      })
+    })
+  }
+
+  # We'll use a pane viwer, and set the minimum height at
+  # 300px to ensure we get enough screen space to display.
+  viewer <- shiny::paneViewer(300)
```

Comment:
I wonder if we'd be better off with a modal? (`dialogViewer`)

Reply:
Switched to a modal now